bin/dktl.sh

```diff
@@ -7,93 +8,137 @@ find-up () {
   echo "$path"
 }
 
-if [ -z `which docker` ]; then
-    echo "You don't seem to have docker installed. Exiting."; exit 1
-fi
-if [ -z `which docker-compose` ]; then
-    echo "You don't seem to have docker-compose installed. Exiting."; exit 1
+# Determine whether we want to run inside the docker container or in the host machine.
+
+if [ -z $DKTL_MODE ] || [ "$DKTL_MODE" = "DOCKER" ]; then
+  DKTL_MODE="DOCKER"
+elif [ "$DKTL_MODE" = "HOST" ]; then
+  DKTL_MODE="HOST"
+else
+  echo "Incorrect DKTL_MODE set to ${DKTL_MODE}. Appropriate values are 'DOCKER'and 'HOST'."
+  exit 1
 fi
 
-dktl_symlink_location=$(which dktl)
-dktl_executable_location=$(readlink $dktl_symlink_location)
+# Check dependencies.
 
-DKTL_DIRECTORY=$(dirname $(dirname $dktl_executable_location))
-export DKTL_DIRECTORY
+if [ "$DKTL_MODE" = "DOCKER" ]; then
+  if [ -z `which docker` ] || [ -z `which docker-compose` ]; then
+      echo "docker and docker-compose are required. Exiting."
+      exit 1
+  fi
+else
+  if [ -z `which php` ] || [ -z `which composer` ] || [ -z `which drush` ]; then
+    echo "php, composer and drush are required. Exiting."
+    exit 1
+  fi
+fi
+
+# Basic Information
 
 DKTL_PROJECT_DIRECTORY=$(find-up dktl.yml)
-if [ -z $DKTL_PROJECT_DIRECTORY ]; then
+if [ -z "$DKTL_PROJECT_DIRECTORY" ]; then
+  if [ "$1" = "init" ]; then
     DKTL_PROJECT_DIRECTORY=$(pwd)
+  else
+    echo "DKTL is running outside of a DKTL project. Run dktl init in the project directory first."
```

Comment:
This doesn't work because if we're running in docker mode, it won't allow us to run `dktl docker-compose` which is required before we can do dktl init. We either need to include `docker:*` (and `dc`) in the conditional or take a different approach.

Reply:
@dafeder if you look at some logic later, when running init, if the containers are not running, the script starts them. It works.